### ISSUE TITLE
iOS: can't access images from local file system

### ISSUE THREAD

**Issue by dxshindeo** [NativeScript/NativeScript#3935]
I took a photo with `nativescript-camera` plugin, which saves the image to the gallery. I then use `nativescript-imagepicker` plugin to open the camera roll and I see the images I took with camera. I select an image, window goes back to the app. I then get the selected images file_uri: `file:///var/mobile/Media/DCIM/100APPLE/IMG_0001.JPG`
But then, when I put it in `Image` tag (dies silently as background-image), console says: `Unhandled Promise rejection: The requested URL was not found on this server.`

So basically this does not show me the image: 
```
<Image src="file:///var/mobile/Media/DCIM/100APPLE/IMG_0001.JPG" style="width: 100; height: 100;"></Image>
```
When I take a picture and then run image-picker functions, I am asked for permissions, which I grant. Also double checked in Settings->my_app->Photos|Camera permissions are given.

NS CLI: 2.5.3
Testing on real iPhone 5.

**Comment by NickIliev** [comment]
Hey @dxshindeo I've tried to reproduce your issue but to no avail.
Keep in mind that you will need explicit permissions for both camera and access to photo gallery which can be requested with the **requestPermissions** method from nativescript-camera plugin. Without the permission granted the camera can take photo but won't be able to save it in the gallery for iOS.

[Here ](https://github.com/NickIliev/NativeScript-Issues-2017/tree/master/NativeScript/issue_3935/CameraTest)is the detailed code I've used to test the camera and the imagepicker working together.

Steps taken:
- clone repo and start application
- tap on **"request permissions"**
- check save to gallery to be active
- take a photo (or photos)
- now use the select multiple to use the image picker
in the album folders (e.g. Camera roll) will open for image selection
- the image is selected successfully and attached via the view-model

**Comment by NickIliev** [comment]
@dxshindeo were you able to resolve this issue? Closing due to inactivity but please let me know if you need further assistance

**Comment by dxshindeo** [comment]
Sorry for the late reply - using your project on my phone I am able to do every action with images - I can see them and etc. 

However, I tried to do the minimum in empty project (--ng) and the same error happens.
Most likely this is an issue with Angular2 + Typescript project setup, because your project is Javascript + Typescript.

Steps to reproduce:
1. `tns create test --ng`
2. Add `nativescript-camera` plugin and full rebuild project.
3. Change files:

```
//app/item/item-detail.component.html

<ActionBar title="Details" class="action-bar"></ActionBar>
<FlexboxLayout flexDirection="column" class="page">
    <FlexboxLayout class="m-15">
        <Label class="h2" [text]="item.id + '. '"></Label>
        <Label class="h2" [text]="item.name"></Label>
    </FlexboxLayout>
    <Label class="h4" [text]="item.role"></Label>


    <Button text="request permission" (tap)="onRequestPermissionsTap()"></Button>
    <Button text="take picture" (tap)="takePicture()"></Button>
    <Image src="{{ image_src }}" style="width: 100; height: 100; background-color: yellow;"></Image>
    <Label text="h4" text="{{ image_src }}"></Label>

</FlexboxLayout>
```

```
//app/item/item-detail.component.ts

import { Component, OnInit } from "@angular/core";
import { ActivatedRoute } from "@angular/router";

import { Item } from "./item";
import { ItemService } from "./item.service";

var camera = require("nativescript-camera");

@Component({
    selector: "ns-details",
    moduleId: module.id,
    templateUrl: "./item-detail.component.html",
})
export class ItemDetailComponent implements OnInit {
    item: Item;
    image_src: any;

    constructor(
        private itemService: ItemService,
        private route: ActivatedRoute
    ) { }

    ngOnInit(): void {
        const id = +this.route.snapshot.params["id"];
        this.item = this.itemService.getItem(id);
    }

    onRequestPermissionsTap() {
        camera.requestPermissions();
    }

    takePicture(args) {
        var dis = this;

        console.log("take pic");
        camera.takePicture({
            width: 100,
            height: 100,
            keepAspectRatio: false,
            saveToGallery: true
        })
        .then(function (imageAsset) {
                console.log(imageAsset);
                dis.image_src = imageAsset;
        }).catch(function (err) {
            console.log("Error -> " + err.message);
        });
    }
}

```

4. Run the app. Select any item. Then press `request permission` to gain permission to access gallery. Then press `take picture`, upon opening you shall get asked for camera permissions, give them.
5. Take a picture, press `use it`, you are back in the app and we get an error in console:
```
ERROR Error: Uncaught (in promise): Error: unsupported URL
```
.. which is being thrown by `Image` tag.

**Comment by viniciodeltoro** [comment]
If I have an image saved in the app 'documents' folder? How can I get the correct path of it and assign it to the 'page.backgroundImage' in iOS?
I tried using 'fs.knownFolders.documents().path' but I get an error when I do that.
The reason I want to use 'page.backgroundImage' instead of an Image element is because I can display the content in 'immersive mode' by hiding the status bar but I'm unable to do that by using an Image element (the screen still shows the status bar area colored with the page's background color)

**Comment by martin-mitro** [comment]
I had the same issue. I had a look in image implementation and I found out that it asks utils.common if path is a file. And there is this code:

`export function isFileOrResourcePath(path: string): boolean {
    if (!types.isString(path)) {
        return false;
    }

    return path.indexOf("~/") === 0 ||  // relative to AppRoot
        path.indexOf("/") === 0 ||      // absolute path
        path.indexOf(RESOURCE_PREFIX) === 0;    // resource
}`

as my stored files start with prefix file:/// this method returns false. So I just removed the prefix so my images start with slash only.

@NickIliev I think it would be nice to update method isFileOrResourcePath. There is also variable FILE_PREFIX in utils-common.ts which is not used.
